File: ployglots/mta-ployglot-pkg-vault/examples/integration/fixtures/env_store_addon.py
"""Example addon for thinkeloquent-fastapi-server consumers.

Signature matches fastapi_server/.agents/addon-author.md from
mta-ployglot-server-bootstrap.
"""
from __future__ import annotations

import os
from typing import Any, Dict

from polyglot_vault_file import EnvStore


def env_store_addon(ctx: Any) -> Dict[str, Any]:
    """Loader addon — call during on_init phase."""
    try:
        result = EnvStore.on_startup(os.environ.get("VAULT_ENV_PATH", ".env"))
        ctx.report.info(f"vault-file: loaded {result.total_vars_loaded} vars")
        return ctx.report.ok(
            {"addon": "env_store", "totalVarsLoaded": result.total_vars_loaded}
        )
    except Exception as err:
        ctx.report.error(err)
        return ctx.report.fail({"addon": "env_store", "message": str(err)})

File: ployglots/mta-ployglot-pkg-vault/examples/integration/fixtures/envStoreAddon.mjs
// Example addon for @ployglot/fastify-server consumers.
// Signature matches fastify_server/.agents/addon-author.md from
// mta-ployglot-server-bootstrap.

import { EnvStore } from '@polyglot/vault-file';

/**
 * @param {object} ctx - fastify addon context with `report` and `app`.
 * @returns {Promise<object>} LoaderReport
 */
export async function envStoreAddon(ctx) {
  try {
    const result = EnvStore.onStartup(process.env.VAULT_ENV_PATH ?? '.env');
    ctx.report.info(`vault-file: loaded ${result.totalVarsLoaded} vars`);
    return ctx.report.ok({ addon: 'envStore', totalVarsLoaded: result.totalVarsLoaded });
  } catch (err) {
    ctx.report.error(err);
    return ctx.report.fail({ addon: 'envStore', message: err.message });
  }
}
